torch/__init__.py

```diff
@@ -287,7 +287,7 @@ def load_shared_libraries(library_path):
 # Appease the type checker; ordinarily this binding is inserted by the
 # torch._C module initialization code in C
 if TYPE_CHECKING:
-    from . import _C as _C
+    from . import _C as _C  # noqa: TCH004
```

Comment:
Also this noqa as I think this is to make the linter happy without affecting import ordering.

Reply:
We can remove the `TYPE_CHECKING` guard around `from torch import _C as _C`. The `_C` submodule is always imported and assigned to `torch._C`.

You can access `torch._C` immediately after `from torch._C import *`

https://github.com/pytorch/pytorch/blob/fb53cd64973167a3f4161d5d48fb11a022bf43f0/torch/__init__.py#L283-L290

The `_C` variable is also being used in the `torch` namespace during importing.

https://github.com/pytorch/pytorch/blob/fb53cd64973167a3f4161d5d48fb11a022bf43f0/torch/__init__.py#L611-L634

Reply:
> You can access `torch._C` immediately after `from torch._C import *`

Here is a minimal example of this:

```console
$ tree foo    
foo
├── __init__.py
├── bar.py
└── baz.py

1 directory, 3 files
```

```python
# foo/__init__.py

from foo.bar import *

assert 'bar' in globals()
print('foo.bar can be accessed after `import foo`')


from .baz import *

assert 'baz' in globals()
print('foo.baz can be accessed after `import foo`')
```

```python
# foo/bar.py

print('foo.bar imported')
```

```python
# foo/baz.py

print('foo.baz imported')
```

```python
In [1]: import foo
foo.bar imported
foo.bar can be accessed after `import foo`
foo.baz imported
foo.baz can be accessed after `import foo`

In [2]: foo.bar
Out[2]: <module 'foo.bar' from '/.../foo/bar.py'>

In [3]: foo.baz
Out[3]: <module 'foo.baz' from '/.../foo/baz.py'>
```

Reply:
Yeah, I think this was specifically to make a linter happy, I agree that from a Python point of view it's fine. I'd like to point out that many linters do not even support '*' imports.

Reply:
I mean remove the `if TYPE_CHECKING` block and add an explicit `from torch._C import _C` at the top level.

`from torch._C import _C` is a side effect of `from torch._C import *` under `torch/__init__.py`. So this change would be safe and make the code more readable.